Add a call ACK / unack example to avoid confusing the retrial mechani… (#11166)
apache/pulsar: site2/docs/concepts-messaging.md

- Messages are acknowledged individually. With individual acknowledgement, the consumer needs to acknowledge each message and sends an acknowledgement request to the broker.
- Messages are acknowledged cumulatively. With cumulative acknowledgement, the consumer only needs to acknowledge the last message it received. All messages in the stream up to (and including) the provided message are not re-delivered to that consumer.

The following example shows how to call the API :

```
consumer.acknowledge(msg);

> ```

Reviewer: Should we also add the example for cumulative acknowledgement here?

Author: My original idea is to let the beginners know that they are going in different ways, because I was confused about this before ....

Reviewer: No problem. So why not add the example for cumulative acknowledgement here? It could be weird that the description contains two ways of acknowledgement but the code only shows the first way.

Reviewer 2: Agree with @BewareMyPower, can you add an example for cumulative acknowledgement? @yonyong

Author: I think you have a point，I have pushed the code @Anonymitaet @BewareMyPower

- Messages are acknowledged individually. With individual acknowledgement, the consumer needs to acknowledge each message and sends an acknowledgement request to the broker.
- Messages are acknowledged cumulatively. With cumulative acknowledgement, the consumer only needs to acknowledge the last message it received. All messages in the stream up to (and including) the provided message are not re-delivered to that consumer.

> The following example shows how to call the API With individual acknowledgement:

Reviewer 2: ```suggestion
If you want to acknowledge messages individually, you can use the following API.
```

Pls keep consistent overall

```java
consumer.acknowledge(msg);
```

> If you are going to acknowledge messages cumulatively, you can use the following API:

Reviewer 2: ```suggestion
If you want to acknowledge messages cumulatively, you can use the following API.
```

Be aware that negative acknowledgment on ordered subscription types, such as Exclusive, Failover and Key_Shared, can cause failed messages to arrive consumers out of the original order.

The following example shows how to call the API :

> ```

Reviewer: ```suggestion
```java
```

Reviewer: Please also apply the change to other code snippets to mark the code block as Java language for correct highlight.

Author: It's done

Be aware that negative acknowledgment on ordered subscription types, such as Exclusive, Failover and Key_Shared, can cause failed messages to arrive consumers out of the original order.

> The following example shows how to call the API :

Reviewer 2: ```suggestion
If you want to acknowledge messages negatively, you can use the following API.
```

                .subscribe();
```

> The following example shows how to call the API to put messages into the retrial queue:

Reviewer 2: ```suggestion
If you want to put messages into a retrial queue, you can use the following API.
```

Author: done @Anonymitaet
